src/components/tx/ApprovalEditor/ApprovalEditorForm.tsx

```diff
@@ -0,0 +1,97 @@
+import PrefixedEthHashInfo from '@/components/common/EthHashInfo'
+import { Grid, Typography, IconButton, SvgIcon, Divider, List, ListItem } from '@mui/material'
+import { useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import css from './styles.module.css'
+import EditIcon from '@/public/images/common/edit.svg'
+import CheckIcon from '@mui/icons-material/Check'
+import type { ApprovalInfo } from '.'
+import classnames from 'classnames'
+import { ApprovalValueField } from './ApprovalValueField'
+import { MODALS_EVENTS } from '@/services/analytics'
+import Track from '@/components/common/Track'
+
+export type ApprovalEditorFormData = {
+  approvals: string[]
+}
+
+export const ApprovalEditorForm = ({
+  approvalInfos,
+  updateApprovals,
+}: {
+  approvalInfos: ApprovalInfo[]
+  updateApprovals?: (newApprovals: string[]) => void
+}) => {
+  const readonly = updateApprovals === undefined
+  const [editIDx, setEditIdx] = useState(-1)
```

Comment:
Nit: can we move `-1` to a constant? It's used in a few places and initially threw me off.